src/redux/rtk/baseQueryWithReauth.api.ts

```diff
@@ -0,0 +1,74 @@
+import { fetchBaseQuery } from '@reduxjs/toolkit/query';
+import type {
+  BaseQueryFn,
+  FetchArgs,
+  FetchBaseQueryError,
+} from '@reduxjs/toolkit/query';
+import { Mutex } from 'async-mutex';
+
+// store
+import { logout } from '../authSlicer';
+
+// services
+import localStorageService from '../../services/local-storage.service';
+
+// models
+import { IAccessDTO } from '../../submodules/public-common/interfaces/dto/auth/iaccess.interface';
+import { BaseRoutes } from '../../submodules/public-common/enums/routes/base-routes.enum';
+import { AuthRoutes } from '../../submodules/public-common/enums/routes/auth-routes.enum';
+import ILoginFullResponse from '../../interfaces/login-full-response';
+
+// constants
+const mutex = new Mutex();
```

Comment:
- [ ] why do we need mutex library?

Reply:
Using async-mutex to prevent multiple calls to '/refreshToken' when multiple calls fail with 401 Unauthorized errors.

https://redux-toolkit.js.org/rtk-query/usage/customizing-queries#preventing-multiple-unauthorized-errors